webapp/test_results_handler.go

```diff
@@ -149,17 +150,13 @@ func getTestRunsAndSources(r *http.Request, runSHA string) (testRunSources []str
 			testRunSources = append(testRunSources, fmt.Sprintf(singleRunURL, afterSpec.Revision, afterSpec.Product.String()))
 		}
 	} else {
-		var sourceURL = `/api/runs?complete=true`
-		if !shared.IsLatest(runSHA) {
-			sourceURL = fmt.Sprintf(`/api/runs?sha=%s`, runSHA)
+		sourceURL, _ := url.Parse("/api/runs")
+		f, err := shared.ParseTestRunFilterParams(r)
+		if err != nil {
+			return nil, nil, err
 		}
-		labels := shared.ParseLabelsParam(r)
-		if labels != nil {
-			for label := range labels.Iterator().C {
-				sourceURL = sourceURL + "&label=" + label.(string)
-			}
-		}
-		testRunSources = []string{sourceURL}
+		sourceURL.RawQuery = f.ToQuery(true).Encode()
+		testRunSources = []string{sourceURL.String()}
```

Comment:
(GitHub doesn't allow me to comment outside of the diff, so I'm commenting here.)

A nit about the comment for this function: it seems we are now (actually, have been) returning an API URL for getting the list of test runs, instead of the list of test runs directly in this function. The documentation suggests otherwise and is probably out of date.

Reply:
Tweaked, but it did say 'array of source urls'.